### PR TITLE
FIX: clean up MediaGallery module

### PR DESCRIPTION
**Pull request by coderimus**
### Description (*)
Provide general clean up for the `MediaGallery` module.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
N/A

### CODE REVIEW

**Review by sivaschenko — CHANGES_REQUESTED** [review]
@coderimus can you please test if the MediaGalleryIndexerTrigger plugin is still functional after this change

**Comment by coderimus** [comment]
@sivaschenko thank you for your review. Yes, a tricky thing eliminated :) I made a fix and push changes. Please, review.

File: MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php [hunk 1]
```diff
@@ -42,8 +42,8 @@ public function __construct(
      */
     public function afterSave(Value $config, Value $result): Value
     {
-        $isMediaGallery = $result->getPath() == self::MEDIA_GALLERY_CONFIG_VALUE;
-        if ($isMediaGallery && $result->isValueChanged() && $result->getValue() == self::MEDIA_GALLERY_ENABLED_VALUE) {
+        $isMediaGallery = $result->getPath() === self::MEDIA_GALLERY_CONFIG_VALUE;
+        if ($isMediaGallery && $result->isValueChanged() && (int) $result->getValue() === self::MEDIA_GALLERY_ENABLED_VALUE) {
```
Comment (hws47a) [inline MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: There is static content error:
```
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 46 | WARNING | Line exceeds 120 characters; contains 126 characters
----------------------------------------------------------------------
```

Comment (hws47a) [inline MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: I personaly would remove a variable and put all checks in one if statement like this
```suggestion
        if ($result->getPath() === self::MEDIA_GALLERY_CONFIG_VALUE 
            && $result->isValueChanged() 
            && (int)$result->getValue() === self::MEDIA_GALLERY_ENABLED_VALUE
        ) {
```

**Review by hws47a — CHANGES_REQUESTED** [review]

File: MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php (hunk 1 above)
Comment (coderimus) [reply MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: @hws47a thank you for your patient to details! I appreciate this.

I made some changes. I moved `isValueChanged` from the if condition. I made this based on the next conclusion: identifying that the path is a media config and that the value changed looks for me as a stand-alone action. So good to keep them separate. What do you think?

Comment (hws47a) [reply MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: I think introducing variables for not complex parts of the code makes it slower to read.

Also for the current case with having a second variable, you call `isValueChanged` even then it's not needed, I mean if `$isMediaGallery` will be false.

Comment (coderimus) [reply MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: Agree. Will provide changes in a few minutes.

Comment (coderimus) [reply MediaGalleryUi/Plugin/MediaGalleryIndexerTrigger.php]: @hws47a thank you for details. I made the recommended changes. Let's wait for the static tests finished and review :)

**Comment by hws47a** [comment]
Hi @coderimus, it seems that Functional tests are failing for your changes. Could you please check them?

**Comment by coderimus** [comment]
Hi @hws47a, I made an investigation and find that at the 1.1-develop the next bug appeared. It possibly caused functional tests failed result. https://github.com/magento/adobe-stock-integration/issues/1144

**Review by hws47a — APPROVED** [review]

**Comment by coderimus** [comment]
@hws47a I merged the latest 1.1-develop and looks like everything is ok :)

**Review by hws47a — APPROVED** [review]

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**